File: app/src/main/java/com/wavesplatform/wallet/crypto/Sha256.java
package com.wavesplatform.wallet.crypto;

import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

public class Sha256 {
    public static byte[] hash(byte[] input) {
        try {
            return MessageDigest.getInstance("SHA-256").digest(input);
        } catch (NoSuchAlgorithmException e) {
            e.printStackTrace();
            throw new IllegalStateException("NoSuchAlgorithmException", e);
        }
    }
}

File: wavesplatform/src/main/java/com/wavesplatform/sdk/crypto/Sha256.kt
package com.wavesplatform.sdk.crypto

import java.security.MessageDigest
import java.security.NoSuchAlgorithmException

class Sha256 {

    companion object {

        fun hash(input: ByteArray): ByteArray {
            try {
                return MessageDigest.getInstance("SHA-256").digest(input)
            } catch (e: NoSuchAlgorithmException) {
                e.printStackTrace()
                throw IllegalStateException("NoSuchAlgorithmException", e)
            }
        }
    }
}
